src/ui.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { Component, ChangeEvent, ReactNode } from "react";
+import { AppBar, Toolbar, Grid, Typography } from "@material-ui/core";
+import { Metadata, MetaItem } from "./search/interfaces";
+import ComboBox from "./search/ComboBox";
+import LabelsAccordion from "./search/LabelsAccordion";
+
+export interface Tile {
+  id: string;
+  name: string;
+  label: string;
+  thumbnail: string; // base64
+}
+interface Props {
+  tiles: Array<Tile>;
+}
+interface State {
+  metadata: Metadata;
+  metadataKeys: string[];
+  imageLabels: string[];
+  imageNames: string[];
+  expanded: string | boolean;
+}
+
+export class UserInterface extends Component<Props, State> {
+  private filteredMeta: Metadata;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      metadata: [],
+      metadataKeys: [],
+      imageLabels: [],
+      imageNames: [],
+      expanded: "labels-toolbox",
+    };
+
+    this.filteredMeta = [];
+  }
+
+  componentDidMount = (): void => {
+    this.loadMeta("metadata.json", (data: Metadata): void => {
+      // Load metadata from json file.
+      if (data && data.length > 0) {
+        this.setState({
+          metadata: data,
+          metadataKeys: Object.keys(data[0]),
+          imageLabels: this.getImageLabels(data),
+          imageNames: this.getImageNames(data),
+        });
+        this.filteredMeta = data;
+      }
+    });
+  };
+
+  handleOnSearchSubmit = (inputKey: string, inputValue: string): void => {
+    if (inputKey === "" || inputValue === "") return;
+
+    // Filter metadata based on inputKey and inputValue
+    const filteredMeta: Metadata = [];
+    this.state.metadata.forEach((mitem: MetaItem) => {
+      for (const [key, value] of Object.entries(mitem)) {
+        if (key === inputKey) {
+          if (
+            (typeof value === "object" && value.includes(inputValue)) ||
+            value === inputValue
+          ) {
+            filteredMeta.push(mitem);
+            break;
+          }
+        }
+      }
```

Comment:
Could this be done without the inner for loop after Craig's typescript fix?

Reply:
Yes, i've changed that in commit `fix: MetaItem is search/interfaces.ts (part2)`.